ShaderTool/Command/Map.cs

```diff
@@ -0,0 +1,266 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using static ShaderTool.Error;
+using static ShaderTool.Util.Util;
+
+namespace ShaderTool.Command {
+
+    class MapData {
+        public string[] actorNames;
+        public string[] materialNames;
+        public string[] textures;
+    }
+
+    class Map {
+
+        public const uint TGR_VERSION = 1;
+
+        public static int MapCommand(string[] args) {
+            AsssertNoneNull(args);
+
+            switch (args[0]) {
+                case "add":
+                    return MapAdd(GetParams(args));
+                case "rm":
+                case "remove":
+                    return MapRm(GetParams(args));
+                case "make":
+                    return MapMake(GetParams(args));
+                case "list":
+                    return MapList();
+                case "addactor":
+                    return MapAddActor(GetParams(args));
+                case "rmactor":
+                case "removeactor":
+                    return MapRmActor(GetParams(args));
+            }
+
+            Console.WriteLine("Wrong parameters! Must be save/add/rm/make/list!");
+            return WRONG_PARAMS;
+
+        }
+
+        public static MapData Load(string mapName) {
+            string resourceFilePath = Program.ResourcesFolder + "\\" + mapName + ".json";
+
+            if (!File.Exists(resourceFilePath))
+                return null;
+
+            string fileContent = File.ReadAllText(resourceFilePath);
+            MapData map = JsonConvert.DeserializeObject<MapData>(fileContent);
+
+            return map;
+        }
+
+        public static void Save(string mapName, MapData map) {
+            string resourceFilePath = Program.ResourcesFolder + "\\" + mapName + ".json";
+            File.WriteAllText(resourceFilePath, JsonConvert.SerializeObject(map, Formatting.Indented));
+        }
+
+        public static void UpdateMaterials(MapData map) {
+
+            List<string> newMaterialNames = new List<string>();
+
+            foreach (string actorName in map.actorNames) {
+
+                string actorPath = Program.ResourcesFolder + @"\" + actorName + @"_Actor.json";
+                string actorFileContent = File.ReadAllText(actorPath);
+                ActorData actor = JsonConvert.DeserializeObject<ActorData>(actorFileContent);
+                string materialName = actor.materialName;
+
+                if (!newMaterialNames.Contains(materialName) && materialName != null) {
+                    newMaterialNames.Add(materialName);
+                }
+
+            }
+
```

Comment:
You missed to update the texture array. foreach material it should check if the texture already exists in the list and append it if neccessary